pkg/xray/resource_xray_watch_test.go

```diff
@@ -0,0 +1,557 @@
+		},
+	})
+}
+
+func TestAccWatch_multipleRepositories(t *testing.T) {
+	_, fqrn, resourceName := mkNames("watch-", "xray_watch")
+	tempStruct := make(map[string]string)
+	copyStringMap(tempStructWatch, tempStruct)
+
+	tempStruct["resource_name"] = resourceName
+	tempStruct["watch_name"] = fmt.Sprintf("xray-watch-%d", randomInt())
+	tempStruct["policy_name_0"] = fmt.Sprintf("xray-policy-%d", randomInt())
+	tempStruct["watch_type"] = "repository"
+	repos := []string{"libs-release-local", "libs-release-local-1"}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+			testAccCreateRepos(t, repos)
+		},
+		CheckDestroy:      verifyDeleted(fqrn, testCheckWatch),
+		ProviderFactories: testAccProviders(),
+		Steps: []resource.TestStep{
+			{
+				Config: executeTemplate(fqrn, multipleRepositoriesWatchTemplate, tempStruct),
+				Check:  verifyXrayWatch(fqrn, tempStruct),
+			},
+		},
+	})
+}
+
+func TestAccWatch_build(t *testing.T) {
+	_, fqrn, resourceName := mkNames("watch-", "xray_watch")
+	tempStruct := make(map[string]string)
+	copyStringMap(tempStructWatch, tempStruct)
+
+	tempStruct["resource_name"] = resourceName
+	tempStruct["watch_name"] = fmt.Sprintf("xray-watch-%d", randomInt())
+	tempStruct["policy_name_0"] = fmt.Sprintf("xray-policy-%d", randomInt())
+	tempStruct["watch_type"] = "build"
+	tempStruct["build_name0"] = "release-pipeline"
+	tempStruct["build_name1"] = "release-pipeline1"
+	builds := []string{tempStruct["build_name0"]}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+			testAccCreateBuilds(t, builds)
+		},
+		CheckDestroy:      verifyDeleted(fqrn, testCheckWatch),
+		ProviderFactories: testAccProviders(),
+		Steps: []resource.TestStep{
+			{
+				Config: executeTemplate(fqrn, buildWatchTemplate, tempStruct),
+				Check:  verifyXrayWatch(fqrn, tempStruct),
+			},
+		},
+	})
+}
+
+func TestAccWatch_multipleBuilds(t *testing.T) {
+	_, fqrn, resourceName := mkNames("watch-", "xray_watch")
+	tempStruct := make(map[string]string)
+	copyStringMap(tempStructWatch, tempStruct)
+
+	tempStruct["resource_name"] = resourceName
+	tempStruct["watch_name"] = fmt.Sprintf("xray-watch-%d", randomInt())
+	tempStruct["policy_name_0"] = fmt.Sprintf("xray-policy-%d", randomInt())
+	tempStruct["watch_type"] = "build"
+	tempStruct["build_name0"] = "release-pipeline"
+	tempStruct["build_name1"] = "release-pipeline1"
+	builds := []string{tempStruct["build_name0"], tempStruct["build_name1"]}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+			testAccCreateBuilds(t, builds)
+		},
+		CheckDestroy:      verifyDeleted(fqrn, testCheckWatch),
+		ProviderFactories: testAccProviders(),
+		Steps: []resource.TestStep{
+			{
+				Config: executeTemplate(fqrn, multipleBuildsWatchTemplate, tempStruct),
+				Check:  verifyXrayWatch(fqrn, tempStruct),
+			},
+		},
+	})
+}
+
+const allReposSinglePolicyWatchTemplate = `resource "xray_security_policy" "security" {
```

Comment:
Merely a comment. I dislike having test setup code that is separate and away from the actual test. Makes it hard to see what the test is about.